core/modules/widgets/slot.js

```diff
@@ -43,13 +43,24 @@ Compute the internal state of the widget
 SlotWidget.prototype.execute = function() {
 	var self = this;
 	this.slotName = this.getAttribute("$name");
-	// Find the parent transclusion
-	var transclusionWidget = this.parentWidget;
-	while(transclusionWidget && !(transclusionWidget instanceof TranscludeWidget)) {
-		transclusionWidget = transclusionWidget.parentWidget;
+	this.slotDepth = parseInt(this.getAttribute("$depth","1"),10) || 1;
+	// Find the parent transclusions
+	var pointer = this.parentWidget,
+		depth = this.slotDepth;
+	while(pointer) {
+		if(pointer instanceof TranscludeWidget) {
+			depth--;
+			if(depth === 0) {
```

Comment:
may be `<== 0` , just to be sure

Reply:
The trouble with doing it that way is that setting a negative depth would incorrectly terminate early. In other words, if somebody tried `<$slot $depth="-100"/>` then it should just fall through to the top, and not find a transclusion widget.

Reply:
OK. So `$depth="-100"` is valid. I thought it wouldn't

Reply:
Hi @pmario setting it to -100 would be a mistake; the important thing is that we don't return the wrong value, nor go into an infinite loop.